Add 476 release notes (#25865)
trinodb/trino: docs/src/main/sphinx/release/release-476.md

* Improve robustness of the OpenX JSON reader when parsing timestamp values. Previously,
  only timestamps with a space separator between date and time were supported. ({issue}`25792`)
* Improve metadata reading performance by optimizing directory listing cache behavior. ({issue}`25749`)
* Fix query failures with `HIVE_CANNOT_OPEN_SPLIT` error when reading ORC files with a large row count. ({issue}`25634`)
* Reduce S3 throttling failures. ({issue}`25781`)
* Fix query failures when attempting to read `date` columns stored as integer
  values in Parquet files. ({issue}`25667`)
* Fix failure when querying views without [StorageDescriptor](https://docs.aws.amazon.com/glue/latest/webapi/API_StorageDescriptor.html) 
  on Glue. ({issue}`25894`)

## Hudi connector

* Add support for user-assigned managed identity authentication for AzureFS. ({issue}`23447`)
* Fix query failures when attempting to read `date` columns stored as integer
  values in Parquet files. ({issue}`25667`)

## Iceberg connector

* Add support for user-assigned managed identity authentication for AzureFS. ({issue}`23447`)
* Add signer type support to the native S3 filesystem. ({issue}`25820`)

> * Add `added_delete_files_count`, `existing_delete_files_count`, `deleted_delete_files_count`, and

Reviewer: ```suggestion
* Add the `added_delete_files_count`, `existing_delete_files_count`, `deleted_delete_files_count`, and
```

* Improve compatibility with S3-compliant storage systems. ({issue}`25791`)
* Show detailed metrics from splits generation in output of `EXPLAIN ANALYZE VERBOSE`. ({issue}`25770`)
* Add `max_partitions_per_writer` session property, which corresponds to the 
  `iceberg.max-partitions-per-writer` configuration property. ({issue}`25662`)
* Improve query planning performance when reading from materialized views. ({issue}`24734`)
* Prevent rare failure when `iceberg.bucket-execution` is enabled. ({issue}`25125`)
* Fix query failures with `HIVE_CANNOT_OPEN_SPLIT` error when reading ORC files with a large row count. ({issue}`25634`)
* Reduce S3 throttling failures. ({issue}`25781`)
* Fix query timeout errors due to concurrent writes on tables with large number of 
  manifest files. ({issue}`24751`)
* Fix query failures when attempting to read `date` columns stored as integer
  values in Parquet files. ({issue}`25667`)
* Fix failure when querying views without [StorageDescriptor](https://docs.aws.amazon.com/glue/latest/webapi/API_StorageDescriptor.html)
  on Glue. ({issue}`25894`)

## Ignite connector

* Improve performance of selective joins for federated queries. ({issue}`25123`)

> ## Kafka Event Listener

Reviewer: ```suggestion
## Kafka event listener
```

* Add feature to deactivate the automated database schema migration for the database backend for resource groups with the property `resource-groups.db-migrations-enabled`. ({issue)`25451`)
* Remove the [](/develop/example-http) from the tar.gz archive and the Docker container.  ({issue}`25128`)
* Fix rare bug when server can hang under load. ({issue}`25816`)
* Require JDK 24 to run Trino. ({issue}`23498`)
* Fix Regression: Graceful shutdown doesn't work. ({issue}`25690`)
* Fix potential query failure when `fault_tolerant_execution_runtime_adaptive_partitioning_enabled` is set to `true`. ({issue}`25870`)

## Security

## Web UI

## JDBC driver

## Docker image

## CLI

## BigQuery connector

* {breaking} Require `--sun-misc-unsafe-memory-access=allow` while running Trino server. ({issue}`25669`)

## Blackhole connector

## Cassandra connector

## ClickHouse connector

* Improve performance of selective joins for federated queries. ({issue}`25123`)

## Delta Lake connector

* Add support for `FOR TIMESTAMP AS OF` clause. ({issue}`21024`)
* Allow configuring User-ManagedIdentity in AzureFS. ({issue}`23447`)
* Add signer type to native S3 filesystem. ({issue}`25820`)

> * Avoid skipping statistics computation for all columns when some column types don't support statistics. ({issue}`24487`)

Reviewer 2: performance, query was not failing

* Improve compatibility with S3-compliant storage systems ({issue}`25791`)
* Improve query planning performance on delta tables. ({issue}`24570`)
* Improve performance when reading tables. ({issue}`25826`)
* Reduce S3 throttling failures by increasing streaming upload part size and reducing number of requests. ({issue}`25781`)
* Fix failure when reading `variant` type column after executing `optimize` procedure. ({issue}`25666`)

> * Fix query failures when attempting to read parquet data with integer logical annotation as a `date` column. ({issue}`25667`)

Reviewer 3: ```
 * Fix `SHOW SCHEMAS `query  failures for catalogs named mixed case ({issue}`25701`)
 ```

* Fix query failure when reading ORC files with a large row count. ({issue}`25634`)
* Reduce S3 throttling failures by increasing streaming upload part size and reducing number of requests. ({issue}`25781`)
* Fix query failures when attempting to read parquet data with integer logical annotation as a `date` column. ({issue}`25667`)

## Hudi connector

* Allow configuring User-ManagedIdentity in AzureFS. ({issue}`23447`)
* Fix query failures when attempting to read parquet data with integer logical annotation as a `date` column. ({issue}`25667`)

## Iceberg connector

* Allow configuring User-ManagedIdentity in AzureFS. ({issue}`23447`)
* Add signer type to native S3 filesystem. ({issue}`25820`)
* Improve compatibility with S3-compliant storage systems ({issue}`25791`)
* Show detailed metrics from splits generation in output of EXPLAIN ANALYZE VERBOSE. ({issue}`25770`)
* Add `max_partitions_per_writer` session property, which corresponds to the `iceberg.max-partitions-per-writer` configuration property. ({issue}`25662`)
* Improve query planning performance when reading from materialized views. ({issue}`24734`)

> * Fix rare failure when `iceberg.bucket-execution` is enabled. ({issue}`25125`)

Reviewer: ```suggestion
* Prevent rare failure when `iceberg.bucket-execution` is enabled. ({issue}`25125`)
```

* Fix query failure when reading ORC files with a large row count. ({issue}`25634`)
* Reduce S3 throttling failures by increasing streaming upload part size and reducing number of requests. ({issue}`25781`)
* Fix query timeout errors due to concurrent writes on tables with large number of manifest files. ({issue}`24751`)
* Fix query failures when attempting to read parquet data with integer logical annotation as a `date` column. ({issue}`25667`)

## Ignite connector

* Improve performance of selective joins for federated queries. ({issue}`25123`)

## JMX connector

## Kafka connector

> ## Kafka Event Listener

Reviewer: I dont think we had that section in the past for any event listeners.. maybe check with old release notes and stay consistent

Reviewer: Fair .. we should add it and other event listerns to the release notes template then.. so we know next time.

> * `kafka-event-listener.client-config-overrides` is removed. To configure the Kafka client for the event listener specify the configuration in a separate file and set `kafka-event-listener.config.resources` to the path to the file. ({issue}`25553`)

Reviewer: ```suggestion
* {{breaking}} Remove the `kafka-event-listener.client-config-overrides` configuration property. To configure the Kafka client for the event listener specify the configuration in a separate file and set `kafka-event-listener.config.resources` to the path to the file. ({issue}`25553`)
```

## Loki connector

## MariaDB connector

* Improve performance of selective joins for federated queries. ({issue}`25123`)

## Memory connector

* Fix failures when deleting rows from memory connector ({issue}`25670`)

## MongoDB connector

## MySQL connector

* Improve performance of selective joins for federated queries. ({issue}`25123`)

## OpenSearch connector

## Oracle connector

* Improve performance of selective joins for federated queries. ({issue}`25123`)
* Improve performance of listing table columns. ({issue}`25231`)

## Pinot connector

## PostgreSQL connector

* Improve performance of selective joins for federated queries. ({issue}`25123`)

## Prometheus connector

## Redis connector

## Redshift connector

* Improve performance of selective joins for federated queries. ({issue}`25123`)

## SingleStore connector

* Improve performance of selective joins for federated queries. ({issue}`25123`)

## Snowflake connector

* Improve performance of selective joins for federated queries. ({issue}`25123`)

> * {{breaking}} Require `--sun-misc-unsafe-memory-access=allow` while running Trino server. ({issue}`25669`)

Reviewer: ```suggestion
* {{breaking}} Require the JVM configuration property `--sun-misc-unsafe-memory-access=allow` to run Trino with the connector. ({issue}`25669`)
```

## SQL Server connector

* Improve performance of selective joins for federated queries. ({issue}`25123`)

## TPC-H connector

## TPC-DS connector

## Vertica connector

* Improve performance of selective joins for federated queries. ({issue}`25123`)

## SPI

> * Block `getSizeInBytes` now returns an estimate of the full data size of the block. ({issue}`25256`)

Reviewer: Something like

Return an estimate of the full data size of the block with `getSizeInBytes`.

# Release 476 (2 Jun 2025)

## General

* Add support for comparing values of `geometry` type. ({issue}`25225`)
* {{breaking}} Require JDK 24 to run Trino. ({issue}`23498`)
* Allow configuring `query.max-memory-per-node` and `memory.heap-headroom-per-node` 
  relative to maximum heap size. ({issue}`25843`)
* Add feature to deactivate the automated database schema migration for the database 
  backend for resource groups with the property `resource-groups.db-migrations-enabled`. ({issue)`25451`)
* Remove the [](/develop/example-http) from the tar.gz archive and the Docker container.  ({issue}`25128`)
* Fix rare bug when server can hang under load. ({issue}`25816`)
* Fix regression that prevented graceful shutdown from working. ({issue}`25690`)
* Fix potential query failure when `fault_tolerant_execution_runtime_adaptive_partitioning_enabled` 
  is set to `true`. ({issue}`25870`)
* Fix failure for queries involving casts with `row` types. ({issue}`25864`)

## Security

## Web UI

## JDBC driver

## Docker image

## CLI

## BigQuery connector

* {{breaking}} Require the `--sun-misc-unsafe-memory-access=allow` JVM configuration 

>   option to run Trino with the connector. ({issue}`25669`)

Reviewer 4: Should we say with `the BigQuery connector`. 
I know it is in the `BigQuery connector` section but I would rather make individual release note entries self descriptive without external context.

## Blackhole connector

## Cassandra connector

## ClickHouse connector

* Improve performance of selective joins for federated queries. ({issue}`25123`)

## Delta Lake connector

* Add support for the `FOR TIMESTAMP AS OF` clause. ({issue}`21024`)

> * Allow configuring *User-ManagedIdentity* in AzureFS. ({issue}`23447`)

Reviewer 4: Add support for user-assigned managed identity authentication for AzureFS ? 

(in all the relevant sections)

> * Add signer type support to the native S3 filesystem. ({issue}`25820`)

Reviewer 4: Allow specifying AWS signing protocol to use for authenticating S3 requests

Author: Thanks. I adjusted it slightly:

> * Allow selecting the AWS signing protocol to use when authenticating S3 requests.

* Avoid skipping statistics computation for all columns when some column types 
  don't support statistics. ({issue}`24487`)
* Improve compatibility with S3-compliant storage systems. ({issue}`25791`)
* Improve query planning performance. ({issue}`24570`)
* Improve performance when reading tables. ({issue}`25826`)
* Reduce S3 throttling failures. ({issue}`25781`)
* Fix failure when reading `variant` type column after executing `optimize` 
  procedure. ({issue}`25666`)
* Fix query failures when attempting to read `date` columns stored as integer
  values in Parquet files. ({issue}`25667`)

> * Fix `SHOW SCHEMAS` failure when catalog names contain mixed case letters. ({issue}`25701`)

Reviewer 4: I think `SHOW SCHEMAS` was just an example of query which failed. The actual class of failed queres was larger.

## Druid connector

* Improve performance of selective joins for federated queries. ({issue}`25123`)

## DuckDB connector

* Improve performance of selective joins for federated queries. ({issue}`25123`)

## Elasticsearch connector

## Exasol connector

* Improve performance of selective joins for federated queries. ({issue}`25123`)

## Faker connector

## Google Sheets connector

* Add support for authentication using delegated user credentials with the 
  `gsheets.delegated-user-email` configuration property. ({issue}`25746`)

## Hive connector

* Add support for excluding certain tables from the directory listing cache 
  with the `hive.file-status-cache.excluded-tables` configuration property. ({issue}`25715`)
* Add signer type support to the native S3 filesystem. ({issue}`25820`)
* Improve compatibility with S3-compliant storage systems. ({issue}`25791`)
* Allow configuring User-ManagedIdentity in AzureFS. ({issue}`23447`)
* Improve the directory listing cache invalidation behavior to avoid invalidating the entire
  cache when dropping a partitioned table. ({issue}`25749`)

> * Fix query failure when reading ORC files with a large row count. ({issue}`25634`)

Reviewer 4: FIx a bug where queries could fail with `HIVE_CANNOT_OPEN_SPLIT` error code when reading tables using ORC file format

Author: Thanks. I adjusted it slightly:

> * Fix query failures with `HIVE_CANNOT_OPEN_SPLIT` error when reading ORC files with a large row count.

> * Reduce S3 throttling failures by increasing streaming upload part size and reducing number

Reviewer 4: make consistent with previous

  of requests. ({issue}`25781`)
* Fix query failures when attempting to read `date` columns stored as integer
  values in Parquet files. ({issue}`25667`)

## Hudi connector

* Allow configuring User-ManagedIdentity in AzureFS. ({issue}`23447`)
* Fix query failures when attempting to read `date` columns stored as integer
  values in Parquet files. ({issue}`25667`)

## Iceberg connector

* Allow configuring User-ManagedIdentity in AzureFS. ({issue}`23447`)
* Add signer type support to the native S3 filesystem. ({issue}`25820`)
* Add `added_delete_files_count`, `existing_delete_files_count`, `deleted_delete_files_count`, and
  `reference_snapshot_id` columns to `$all_manifests` metadata tables. ({issue}`25867`)
* Improve compatibility with S3-compliant storage systems. ({issue}`25791`)
* Show detailed metrics from splits generation in output of `EXPLAIN ANALYZE VERBOSE`. ({issue}`25770`)

> * Add `max_partitions_per_writer` session property, which corresponds to the 

Reviewer 4: are we linking to the docs from release notes if we reference a config property? (not sure if there is actually any documentation for that to be fair).

  `iceberg.max-partitions-per-writer` configuration property. ({issue}`25662`)
* Improve query planning performance when reading from materialized views. ({issue}`24734`)

> * Prevent rare failure when `iceberg.bucket-execution` is enabled. ({issue}`25125`)

Reviewer 4: Fix occasional query failures when `iceberg.bucket-execution` was enabled.

* Fix query failure when reading ORC files with a large row count. ({issue}`25634`)

> * Reduce S3 throttling failures by increasing streaming upload part size and reducing 

Reviewer 4: `Reduce S3 throttling failures`

  number of requests. ({issue}`25781`)

> * Fix query timeout errors due to concurrent writes on tables with large number of 

Reviewer 4: @raunaqmorarka what `writes` mean hear. Looks like the PR is about some lock contention, right?

Reviewer 2: Writes in iceberg update some metadata at the end in the coordinator, this operation was earlier using the same threadpool as the one used by query planning and blocking those if the write was large enough. Now we separated the threadpool for query planning.
